gradle-jdks-setup/src/main/java/org/gradle/wrapper/GradleWrapperMain.java

```diff
@@ -0,0 +1,211 @@
+        Class<?> origGradleWrapperMainClass = loader.loadClass("org.gradle.wrapper.OrigGradleWrapper");
+        Method method = origGradleWrapperMainClass.getMethod("main", String[].class);
+        method.invoke(null, new Object[] {args});
+    }
+
+    private static void configureJdkAutomanagement() throws IOException {
+        // Delegate to gradle-jdks-setup.sh to install the JDK
+        // TODO(crogoz): maybe use java to do the setup
+        Path projectHome = projectHome();
+        CommandRunner.run(List.of("sh", "./gradle/gradle-jdks-setup.sh"), Optional.of(projectHome.toFile()));
+
+        // Set the daemon Java Home
+        Path jdkMajorVersionPath = projectHome.resolve("gradle/gradle-jdk-major-version");
+        String majorVersion = Files.readString(jdkMajorVersionPath).trim();
+        Path localPathFile = projectHome
+                .resolve("gradle/jdks")
+                .resolve(majorVersion)
+                .resolve(getOs())
+                .resolve(getArch())
+                .resolve("local-path");
+        String localJdkFileName = Files.readString(localPathFile).trim();
+        Path jdkInstallationPath = getGradleJdksPath().resolve(localJdkFileName);
+        System.out.println("Setting daemon Java Home to " + jdkInstallationPath.toAbsolutePath());
+        System.setProperty(
+                "org.gradle.java.home", jdkInstallationPath.toAbsolutePath().toString());
+
+        // Disable auto-download and auto-detect of JDKs
+        // System.setProperty("org.gradle.java.installations.auto-download", "false");
+        // System.setProperty("org.gradle.java.installations.auto-detect", "false");
+
+        // TODO(crogoz): read all jdks installations & set them up
+    }
+
+    private static Path projectHome() {
+        return wrapperJar().toAbsolutePath().getParent().getParent().getParent();
+    }
+
+    private static Path wrapperJar() {
+        URI location;
+        try {
+            location = GradleWrapperMain.class
+                    .getProtectionDomain()
+                    .getCodeSource()
+                    .getLocation()
+                    .toURI();
+        } catch (URISyntaxException var3) {
+            throw new RuntimeException(var3);
+        }
+
+        if (!location.getScheme().equals("file")) {
+            throw new RuntimeException(
+                    String.format("Cannot determine classpath for wrapper Jar from codebase '%s'.", location));
+        } else {
+            try {
+                return Paths.get(location);
+            } catch (NoClassDefFoundError var2) {
+                return Paths.get(location.getPath());
+            }
+        }
+    }
+
+    private static Path getGradleJdksPath() {
+        return Path.of(Optional.ofNullable(System.getenv("GRADLE_USER_HOME"))
+                        .orElseGet(() -> System.getProperty("user.home") + "/.gradle"))
+                .resolve("gradle-jdks");
+    }
+
+    public static String getArch() {
```

Comment:
These are the same as `CurrentOs` and `CurrentArch` etc right? Eventually we shouldn't need these in the Gradle plugin (only in the setup jar) after we delete the in-gradle downloading of JDKs, but until then they should probably live in some common jar with `CommandRunner` to avoid duplicating the logic.

Reply:
Yes, I'll need to move these around to avoid duplication. I also need Os & Arch classes as well.